### PR TITLE
feat(semver): optional SemVer build and prerelease properties

### PR DESCRIPTION
**Pull request by timreichen**
makes `SemVer` `prerelease` and `build` properties optional to reduce empty array boilerplate.

### CODE REVIEW

File: semver/increment.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export function increment(
     // If the input is a non-prerelease version, this acts the same as
     // prepatch.
     case "prerelease":
-      if (version.prerelease.length === 0) {
+      if (version.prerelease?.length === 0) {
```
Comment (kt3k) [inline semver/increment.ts:110]: When `prerelease` is `undefined`, that should be considered as the same as empty array. So this line seems introducing a bug. (this passes with prerelease = `[]`, but doesn't pass with prerelease = `undefined`)

Can you add test cases to `increment_test.ts` to exercise these patterns?

Comment (kt3k) [reply semver/increment.ts:110]: Probably we need to translate `version.prerelease.length === 0` to `version.prerelease === undefiend || version.prerelease.length === 0`

File: semver/is_semver_test.ts [hunk 2]
```diff
@@ -47,6 +45,8 @@ Deno.test({
   fn: async (t) => {
     let i = 0;
     const versions: [unknown][] = [
+      [{ major: 0, minor: 0, patch: 0, prerelease: [] }],
+      [{ major: 0, minor: 0, patch: 0, build: [] }],
```
Comment (kt3k) [inline semver/is_semver_test.ts:49]: We also need to add test cases to check with both prerelease and build undefined

File: semver/increment.ts [hunk 3]
```diff
@@ -134,7 +134,7 @@ export function increment(
       if (
         version.minor !== 0 ||
         version.patch !== 0 ||
-        version.prerelease.length === 0
+        version.prerelease?.length === 0
```
Comment (kt3k) [inline semver/increment.ts:137]: ditto. This seems introducing a bug

File: semver/increment.ts [hunk 4]
```diff
@@ -161,7 +161,7 @@ export function increment(
       // 1.2.1 bumps to 1.3.0
       if (
         version.patch !== 0 ||
-        version.prerelease.length === 0
+        version.prerelease?.length === 0
```
Comment (kt3k) [inline semver/increment.ts:164]: Ditto. This seems introducing a bug. Please add test case to exercise this pattern with prerelease = `undefined`.

File: semver/increment.ts [hunk 5]
```diff
@@ -186,7 +186,7 @@ export function increment(
       // If it is a pre-release it will bump up to the same patch version.
       // 1.2.0-5 patches to 1.2.0
       // 1.2.0 patches to 1.2.1
-      if (version.prerelease.length === 0) {
+      if (version.prerelease?.length === 0) {
```
Comment (kt3k) [inline semver/increment.ts:189]: Ditto. This seems introducing a bug. Please add test case to exercise this pattern with prerelease = `undefined`.

File: semver/is_semver_test.ts (hunk 2 above)
Comment (timreichen) [reply semver/is_semver_test.ts:50]: done

File: semver/increment.ts (hunk 1 above)
Comment (timreichen) [reply semver/increment.ts]: done

File: semver/increment.ts (hunk 3 above)
Comment (timreichen) [reply semver/increment.ts]: fixed

File: semver/increment.ts (hunk 4 above)
Comment (timreichen) [reply semver/increment.ts]: fixed

File: semver/increment.ts (hunk 5 above)
Comment (timreichen) [reply semver/increment.ts]: done

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**